### PR TITLE
ts-migration/convert-some-utils-functions

### PR DESCRIPTION
**Pull request by G-Rath**
This might cause issues with #310 but these three functions are the only ones from `tsUtils` that are required for about three or four rules, so getting these across the line will let me land those straight away :)

### CODE REVIEW

**Comment by SimenB** [comment]
Coverage isn't happy... Would it be quick to port a rule which uses these?

**Comment by SimenB** [comment]
I've merged #310 as well, fwiw. Doesn't seem like it created a merge conflict 🙂

**Comment by G-Rath** [comment]
@SimenB done - I've pushed `no-jasmine-globals` :)

File: src/rules/no-jasmine-globals.ts [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
-export default {
+export default createRule({
+  name: __filename,
   meta: {
     docs: {
-      url: getDocsUrl(__filename),
+      category: 'Best Practices',
+      description: 'Disallow Jasmine globals',
+      recommended: false,
```
Comment (SimenB) [inline src/rules/no-jasmine-globals.ts]: This is recommended :)

File: src/rules/no-jasmine-globals.ts [hunk 2]
```diff
@@ -109,18 +117,26 @@ export default {
...
-                fix(fixer) {
-                  return [
-                    fixer.replaceText(
-                      node.parent,
-                      `jest.setTimeout(${node.parent.right.value})`,
-                    ),
-                  ];
-                },
+                fix:
+                  right.type !== AST_NODE_TYPES.Literal
```
Comment (SimenB) [inline src/rules/no-jasmine-globals.ts]: Can you flip it so the `true` case is first in the ternary? Or do a `return` above this if it's of the wrong type

**Comment by G-Rath** [comment]
@SimenB this is good to go!

I pulled in another rule (`no-disabled-tests`) to help improve the coverage, along w/ adding a couple more tests & re-arranging some type-narrowing if conditions so that they get hit before non-narrowing ones :)

Everything's now back at 100% coverage, and `utils.js` has gotten smaller since those two rules were the only ones using some of the methods :D

File: src/rules/__tests__/no-jasmine-globals.test.ts [hunk 3]
```diff
@@ -53,6 +53,10 @@ ruleTester.run('no-jasmine-globals', rule, {
       errors: [{ messageId: 'illegalJasmine', column: 1, line: 1 }],
       output: 'jest.setTimeout(5000);',
     },
+    {
+      code: 'jasmine.DEFAULT_TIMEOUT_INTERVAL = function() {}',
```
Comment (SimenB) [inline src/rules/__tests__/no-jasmine-globals.test.ts]: is this valid jasmine code?

Comment (SimenB) [reply src/rules/__tests__/no-jasmine-globals.test.ts]: I'm happy to warn on it either way

Comment (G-Rath) [reply src/rules/__tests__/no-jasmine-globals.test.ts]: I don't think so, but it's technically something that could be done. Eitherway, it triggers the previously uncovered branch :)

**Pull request merged**